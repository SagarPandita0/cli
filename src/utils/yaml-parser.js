import frontMatter from 'front-matter'
import { readFile } from 'fs/promises'

export async function convertToJson(filePath) {
  const fileContent = await readFile(filePath, 'utf8')
  const json = frontMatter(fileContent)
  return json
}
