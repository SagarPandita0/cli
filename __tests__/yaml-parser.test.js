import { URL } from 'url'
import path from 'path'

import { convertToJson } from '../src/utils/yaml-parser.js'

const __dirname = new URL('.', import.meta.url).pathname

test('convert markdown file to json', async () => {
  const filePath = path.join(__dirname, './__fixtures__/datafiles/event-ok.md')
  const jsonResult = await convertToJson(filePath)
  expect(jsonResult).toMatchSnapshot()
})
