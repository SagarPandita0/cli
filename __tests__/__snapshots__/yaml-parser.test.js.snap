// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convert markdown file to json 1`] = `
Object {
  "attributes": Object {
    "city": "Ra'anana",
    "country": "Israel",
    "country_code": "IL",
    "date": 2020-02-18T00:00:00.000Z,
    "images": Array [
      "https://pbs.twimg.com/media/EO8sCpbWAAAUAw0?format=jpg&name=4096x4096",
    ],
    "language": "English",
    "name": "SAP Labs dkom 2020 Israel",
    "recognitions": Object {
      "twitter": Array [
        "https://twitter.com/liran_tal/status/1220242496703385600?s=20",
      ],
    },
    "recording_url": null,
    "slides_url": null,
    "tags": "post",
    "title": "The State of Open Source Security",
    "type": "conference",
    "url": null,
  },
  "body": "",
  "bodyBegin": 20,
  "frontmatter": "date: 2020-02-18
tags: post
name: SAP Labs dkom 2020 Israel
url:
type: conference
title: The State of Open Source Security
slides_url:
recording_url:
city: Ra'anana
country: Israel
country_code: IL
language: English
recognitions:
  twitter:
    - https://twitter.com/liran_tal/status/1220242496703385600?s=20
images:
  - https://pbs.twimg.com/media/EO8sCpbWAAAUAw0?format=jpg&name=4096x4096",
}
`;
